### DISCUSSION TITLE
"memory access out of bounds" on allocating large structs.

### DISCUSSION THREAD

**Discussion by karikera** [wasm-bindgen/wasm-bindgen#3986] [Q&A]
### Describe the Bug

memory access out of bounds on allocating large structs.

runtime error

wasm-bindgen@0.2.92

### Steps to Reproduce
```rs

#[wasm_bindgen]
pub struct FoobarPass {
    buf:[i32; 1024*512],
}

#[wasm_bindgen]
pub fn foobar_pass()->FoobarPass { // it's okay
    FoobarPass{
        buf: [0; 1024*512],
    }
}

#[wasm_bindgen]
pub struct FoobarFail {
    buf:[i32; 1024*256],
    n:i64,
}

#[wasm_bindgen]
pub fn foobar_fail()->FoobarFail { // it fails
    FoobarFail{
        buf: [0; 1024*256],
        n:0,
    }
}

```

### Expected Behavior

no error

### Actual Behavior

Uncaught RuntimeError RuntimeError: memory access out of bounds

**Comment by daxpedda** [comment]
I didn't investigate the exact issue here, but its generally not a good idea to allocate a whole MB on the stack.
You might want to consider using a `Vec` here instead.
